frontend/src/scenes/experiments/ExperimentView/VariantPreviewImage.tsx

```diff
@@ -0,0 +1,117 @@
+import { IconUpload, IconX } from '@posthog/icons'
+import { LemonButton, LemonDivider, LemonFileInput, LemonModal, LemonSkeleton, lemonToast } from '@posthog/lemon-ui'
+import { useActions, useValues } from 'kea'
+import { useUploadFiles } from 'lib/hooks/useUploadFiles'
+import { useState } from 'react'
+
+import { experimentLogic } from '../experimentLogic'
+import { VariantTag } from './components'
+
+export function VariantPreviewImage({
+    variantKey,
+    rolloutPercentage,
+}: {
+    variantKey: string
+    rolloutPercentage: number
+}): JSX.Element {
+    const { experiment } = useValues(experimentLogic)
+    const { updateExperimentVariantImages } = useActions(experimentLogic)
+
+    const [mediaId, setMediaId] = useState(experiment.parameters?.variant_preview_media_ids?.[variantKey] || null)
+    const [isLoadingImage, setIsLoadingImage] = useState(true)
+    const [isModalOpen, setIsModalOpen] = useState(false)
+
+    const { setFilesToUpload, filesToUpload, uploading } = useUploadFiles({
+        onUpload: (_, __, id) => {
+            setMediaId(id)
+            if (id) {
+                const updatedVariantImages = {
+                    ...experiment.parameters?.variant_preview_media_ids,
+                    [variantKey]: id,
+                }
+                updateExperimentVariantImages(updatedVariantImages)
+            }
+        },
+        onError: (detail) => {
+            lemonToast.error(`Error uploading image: ${detail}`)
+        },
+    })
+
+    return (
+        <div className="space-y-4">
+            {!mediaId ? (
+                <LemonFileInput
+                    accept="image/*"
+                    multiple={false}
+                    onChange={setFilesToUpload}
+                    loading={uploading}
+                    value={filesToUpload}
+                    callToAction={
+                        <div className="relative">
+                            <div className="Lettermark Lettermark--xlarge Lettermark--rounded Lettermark--variant-4">
+                                <IconUpload />
+                            </div>
+                        </div>
+                    }
+                />
+            ) : (
+                <div className="relative">
+                    <>
+                        <div className="text-muted inline-flex flow-row items-center gap-1 cursor-pointer">
+                            <div onClick={() => setIsModalOpen(true)} className="cursor-pointer relative">
```

Comment:
I had no idea that I must click on this tile to see the preview in a large scale – only realized this looking at the code! In other file upload contexts you click on the image to _change_ it (e.g. org logo or Hog destinations). But an easy way to create the affordance for image expansion is `cursor: zoom-in`:

<img width="85" alt="Screenshot 2024-10-07 at 12 21 55" src="https://github.com/user-attachments/assets/a619c8f9-3829-41c1-89ab-87a1ed1784c2">